internal/pipeline/executor.go

```diff
@@ -281,12 +281,13 @@ func adjustRelPath(resources []*yaml.RNode, relPath string) ([]*yaml.RNode, erro
 		return resources, nil
 	}
 	for _, r := range resources {
-		meta, err := r.GetMeta()
+		currPath, _, err := kioutil.GetFileAnnotations(r)
 		if err != nil {
 			return resources, err
 		}
-		if !strings.HasPrefix(meta.Annotations[kioutil.PathAnnotation], relPath) {
-			newPath := path.Join(relPath, meta.Annotations[kioutil.PathAnnotation])
+		// if currPath is relative to root pkg i.e. already has relPath, skip it
+		if !strings.HasPrefix(currPath, relPath+"/") {
```

Comment:
Why use HasPrefix instead of a function in path package?

Reply:
Unfortunately, I couldn't find a method from path that can help. I think there is probably better way to solve this, so the problem we have is:
I have a relative path say relpath = `a/b` (note - without the slash) here and I have a path say currPath = `a/b/c/d` and want to find out if currPath is relative to relPath.

Reply:
I tried to see if there is a direct method in path to do that easily. Closest thing we can use is `filepath.Rel()` but it is not sufficient.